UI/streaming-settings-util.cpp

```diff
@@ -0,0 +1,211 @@
+	if (CheckInMapAndSelected(map, kSettingsResponseKeys.framerate)) {
+		double currentFPS = CommonSettings::GetConfigFPSDouble(config);
+		double newFPS =
+			doubleFromMap(map, kSettingsResponseKeys.framerate);
+		if (abs(currentFPS - newFPS) >
+		    0.001) { // Only change if different
+			if (abs(floor(newFPS) - newFPS) >
+			    0.01) { // Is a drop-frame FPS
+				int num = ceil(newFPS) * 1000;
+				int den = 1001;
+				config_set_uint(config, "Video", "FPSType",
+						2); // Fraction
+				config_set_uint(config, "Video", "FPSNum", num);
+				config_set_uint(config, "Video", "FPSDen", den);
+			} else { // Is integer FPS
+				config_set_uint(config, "Video", "FPSType",
+						1); // Integer
+				config_set_uint(config, "Video", "FPSInt",
+						(int)floor(newFPS));
+			}
+		}
+	}
+
+	if (CheckInMapAndSelected(map, kSettingsResponseKeys.videoBitrate)) {
+		int newBitrate =
+			intFromMap(map, kSettingsResponseKeys.videoBitrate);
+		config_set_int(config, "SimpleOutput", "VBitrate", newBitrate);
+	}
+
+	if (CheckInMapAndSelected(map, kSettingsResponseKeys.h264Profile)) {
+		QString profile =
+			stringFromMap(map, kSettingsResponseKeys.h264Profile);
+		profile = profile.toLower();
+		x264SettingList.append("profile=" + profile);
+	}
+
+	if (CheckInMapAndSelected(map, kSettingsResponseKeys.h264Level)) {
+		QString levelString =
+			stringFromMap(map, kSettingsResponseKeys.h264Level);
+		x264SettingList.append("level=" + levelString);
+	}
+
+	if (CheckInMapAndSelected(map, kSettingsResponseKeys.gopSizeInFrames)) {
+		int gopSize =
+			intFromMap(map, kSettingsResponseKeys.gopSizeInFrames);
+		x264SettingList.append("keyint=" + QString::number(gopSize));
+	}
+
+	if (CheckInMapAndSelected(map, kSettingsResponseKeys.gopClosed)) {
+		bool gopClose =
+			boolFromMap(map, kSettingsResponseKeys.gopClosed);
+		if (gopClose) {
+			x264SettingList.append("open_gop=0");
+		} else {
+			x264SettingList.append("open_gop=1");
+		}
+	}
+
+	if (CheckInMapAndSelected(map, kSettingsResponseKeys.gopBFrames)) {
+		int bFrames = intFromMap(map, kSettingsResponseKeys.gopBFrames);
+		x264SettingList.append("bframes=" + QString::number(bFrames));
+	}
+
+	if (CheckInMapAndSelected(map, kSettingsResponseKeys.gopRefFrames)) {
+		int refFrames =
+			intFromMap(map, kSettingsResponseKeys.gopRefFrames);
+		x264SettingList.append("ref=" + QString::number(refFrames));
+	}
+
+	// kSettingsResponseKeys.streamRateControlMode defaults to CBR in Simple
+	// encoder mode. Can add later for advanced panel.
+
+	if (CheckInMapAndSelected(map,
+				  kSettingsResponseKeys.streamBufferSize)) {
+		int bufferSize =
+			intFromMap(map, kSettingsResponseKeys.streamBufferSize);
+		x264SettingList.append("vbv-bufsize=" +
+				       QString::number(bufferSize));
+	}
+
+	QString x264String = x264SettingList.join(" ");
+	const char *x264_c_String = x264String.toStdString().c_str();
+	config_set_string(config, "SimpleOutput", "x264Settings",
+			  x264_c_String);
```

Comment:
For me, on VS2019 on Windows 10, this returns a garbage string.  VS2019 gives me the warning:
> C26815
> The pointer is dangling because it points at a temporary instance which was destroyed.

This works:
```
	std::string x264String = x264SettingList.join(" ").toStdString();
	const char *x264_c_String = x264String.c_str();
	config_set_string(config, "SimpleOutput", "x264Settings",
			  x264_c_String);
```

Alternatively, the pattern of assigning to a `std::string` with `toUtf8().constData()` is used throughout the OBS code:
```
	std::string x264String = x264SettingList.join(" ").toUtf8().constData();
	const char *x264_c_String = x264String.c_str();
	config_set_string(config, "SimpleOutput", "x264Settings",
			  x264_c_String);
```

Reply:
> Is UI/pre-stream-wizard/CMakeLists.txt necessary to include?

Cmake requires a file, even if blank, for every sour directory 

Thank you for the edits. Adding now. for some reason `page-input-display.cpp` notes won't load no matter how i poke at github lol